sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client_options.hpp

```diff
@@ -0,0 +1,224 @@
+   * {@link AttestOptions::RuntimeData} and {@link AttestOptions::InittimeData} fields.
+   */
+  class AttestationDataType final {
+  private:
+    std::string m_dataType;
+
+  public:
+    /**
+     * @brief Construct a new attestation DataType object
+     *
+     * @param dataType The string version for the Key Vault keys service.
+     */
+    AttestationDataType(std::string dataType) : m_dataType(std::move(dataType)) {}
+    AttestationDataType() {}
+    /**
+     * @brief Enable comparing the ext enum.
+     *
+     * @param other Another AttestationDataType to be compared.
+     */
+    bool operator==(AttestationDataType const& other) const
+    {
+      return m_dataType == other.m_dataType;
+    }
+
+    /**
+     * @brief Return the #AttestationDataType string representation.
+     *
+     */
+    std::string const& ToString() const { return m_dataType; }
+
+    /**
+     * @brief When specified, instructs the attestation service to express the runtime data in the
+     * generated token as a JSON object.
+     *
+     */
+    AZ_ATTESTATION_DLLEXPORT static const AttestationDataType Json;
+
+    /**
+     * @brief When specified, instructs the attestation service to express the runtime data in the
+     * generated token as a Binary object.
+     *
+     */
+    AZ_ATTESTATION_DLLEXPORT static const AttestationDataType Binary;
+  };
+
+  /** @brief AttestationData represents a block of data to be sent to the attestation service.
+   * See the description of the {@link AttestationClient} class for more information about how the
+   * AttestationData type works.
+   */
+  struct AttestationData final
+  {
+    /// Data contained within attestation evidence. The attestation service will verify that the
+    /// evidence does contain this data and will include the Data in the attestation token. A
+    /// relying party can then use this data.
+    std::vector<uint8_t> Data;
+
+    /// Reflects how the Data field should be represented in the resulting attestation token.
+    AttestationDataType DataType;
+  };
+
+  /** @brief Parameters sent to the attestation service to be consumed in an attestation operation.
+   */
+  struct AttestOptions final
+  {
+    /// Data created dynamically by the enclave
+    Azure::Nullable<AttestationData> RuntimeData{};
+
+    /// Data created when the enclave is created.
+    Azure::Nullable<AttestationData> InittimeData{};
+
+    /// Nonce which is sent to the attestation service to allow a caller to prevent replay attacks.
+    Azure::Nullable<std::string> Nonce{};
+
+    /// A test hook which allows developers to test attestation policies before they commit them to
+    /// the service.
+    Azure::Nullable<std::string> DraftPolicyForAttestation{};
+
+    /// Specifies the options which should be used to validate the attestation token returned
+    /// by the attestation service. If not provided by the caller, the token validation options
```

Comment:
/**
     * @brief Define policies to be called one time for every HTTP request from an SDK client.
     *
     */